### ISSUE TITLE
[Question] About input parameters for get_ospf_interfaces

### ISSUE THREAD

**Issue by ksaegusa** [CiscoTestAutomation/genielibs#57]
`get_ospf_interfaces`

Why is the AS number of BGP required in getting the configuration of OSPF interface?

There seemed to be no difference in the results of the acquisition whether I left it blank or put in a value.

https://github.com/CiscoTestAutomation/genielibs/blob/fac90539e156cafdd8054d3def94af639b048442/pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/ospf/get.py#L492

https://github.com/CiscoTestAutomation/genielibs/blob/fac90539e156cafdd8054d3def94af639b048442/pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/ospf/get.py#L510

It seems to me that the insntans key is not included in 'parse("show ip ospf interface brief")'

Is ospf_process_id correct?
```
def get_ospf_interfaces(device, ospf_process_id ):
    """ Retrieve interface for ospf using ospf_process_id 
        Args:
            device ('obj'): Device object
            ospf_process_id ('int'): OSPF Process ID
        Returns:
            List of interfaces
        Raises:
            SchemaEmptyParserError
    """
    try:
        out = device.parse("show ip ospf interface brief")
    except SchemaEmptyParserError:
        return None

    try:
        areas_dict = out["instance"][ospf_process_id ]['areas']
    except KeyError:
        return None
```

**Comment by ksaegusa** [comment]
Modify type
```
def get_ospf_interfaces(device, ospf_process_id ):
    """ Retrieve interface for ospf using ospf_process_id 
        Args:
            device ('obj'): Device object
            ospf_process_id ('int'): OSPF Process ID
        Returns:
            List of interfaces
        Raises:
            SchemaEmptyParserError
    """
    try:
        out = device.parse("show ip ospf interface brief")
    except SchemaEmptyParserError:
        return None

    try:
        areas_dict = out["instance"][str(ospf_process_id) ]['areas']
    except KeyError:
        return None
```

**Comment by tahigash** [comment]
Thanks for pointing out. yes, you are correct. it should be `ospf_process_id`. but someone might be already using it. so we need to have backward compatibility. I will find the fix and put the fix in next release 21.6 (end of this month)

**Comment by tahigash** [comment]
The API has been fixed internally and the fix will be released with 21.6 as I mentioned above. Closing.
